lib/resources/orders.js

```diff
@@ -391,6 +391,83 @@ const updateOrderItemsInContext = orderContext => {
   };
 };
 
+/**
+ * Cancel an order function.
+ *
+ * @typedef module:resources/orders~CancelFunc
+ *
+ * @property {string} orderOd order ID.
+ * @property {object} [params] [Request]{@link https://github.com/request/request} params to pass to the API call.
+ * @property {module:api~requestCallback} callback Callback that handles the response.
+ *
+ * @example
+ * // Make a post request to the API for the order with the ID of 'abc123' for program 12
+ * ro.program(12).order.cancel({
+ *  orderId: 'abc123',
+ *  refundReasonDescription: 'reason',
+ *  params: {
+ *    // params
+ *  },
+ *  callback: (error, data) => {
+ *     // ...
+ *  }
+ * });
+ */
+
+/**
+ * Higher order function for creating a `cancelOrder` function in context.
+ *
+ * @param {module:resources/orders~OrderContext} orderContext Order context object
+ *
+ * @returns {module:resources/orders~CancelFunc} `cancelOrder` function in context
+ *
+ * @see {@link module:resources/orders~CancelFunc} for examples.
+ *
+ * @protected
+ */
+const cancelOrder = orderContext => {
+  return function cancel({ orderId, refundReasonDescription, params, callback }) {
+    let error;
+    let errorMessage;
+
+    // // Argument validations.
+    if (typeof params !== 'object') {
+      // Validate that `params` exists and is an object.
+      errorMessage = 'A params object is required';
+    } else if (config.get('apiVersion') === 'v4' && (!orderId || typeof orderId !== 'string')) {
+      // Validate that `orderId` exists.
+      // NOTE: This is only required in RewardOps API v4
+      errorMessage = 'must pass an orderId to `orders.cancel()`';
+    } else if (
+      config.get('apiVersion') === 'v4' &&
```

Comment:
Comment: So Devin confirmed that v3 is not supported. There is no v5 version of program refund.

Should we only accept v4?

Reply:
That's what I think 🤔

Reply:
Should we add another if/else statement? E.g.

```
if (typeof params !== 'object') {
      // Validate that `params` exists and is an object.
      errorMessage = 'A params object is required';
    } else if (config.get('apiVersion') !== 'v4') {
      errorMessage = 'cancel order is only supported in v4';
    }
  ...
```